dockers/README.md

```diff
@@ -0,0 +1,223 @@
+
+For first time users of these images it is helpful to have a README which gives a high-level overview of the image.
+
+See the examples for samtools([docker_build](./broad/samtools/docker_build.sh), [docker_versions](./broad/samtools/docker_versions.tsv), [README](./broad/samtools/README.md))
+
+## Formatting
+
+Formatting our Dockerfiles consistenty helps improve readability and eases maintenance headaches down the road. The following are a couple of tenants that we follow when writing our Dockerfiles:
+
+* ARGS, ENV, LABEL in that order
+* Always add versions of tools in the LABEL
+* Single RUN steps
+* Alphabetize package install
+* Clean up package index cache
+* Use ; instead of && for line continuation
+* Logically seperate steps within RUN
+* Four spaces per tab indent
+* Short comments to describe each step
+* tini is always default entrypoint
+
+The following is a good example for our `verify_bam_id` image. This Dockerfile shows how to install packages, install tini and clean up cached index files for a debian base image.
+
+##### :eyes: Example
+```dockerfile
+# Have to use debian based image, many of the installed packages here are not available in Alpine
+FROM us.gcr.io/broad-dsp-gcr-public/base/python:debian
+
+ARG GIT_HASH=c1cba76e979904eb69c31520a0d7f5be63c72253
+
+ENV TERM=xterm-256color \
+    BAMID_URL=https://github.com/Griffan/VerifyBamID/archive \
+    TINI_VERSION=v0.19.0
+
+LABEL MAINTAINER="Broad Institute DSDE <dsde-engineering@broadinstitute.org>" \
+        GIT_HASH=${GIT_HASH}
+
+WORKDIR /usr/gitc
+
+# Install dependencies
+RUN set eux; \
+        apt-get update; \
+        apt-get install -y \ 
+            autoconf \
+            cmake \
+            g++ \
+            gcc \
+            git \
+            libbz2-dev \        
+            libcurl4-openssl-dev \
+            libhts-dev \
+            libssl-dev  \
+            unzip \
+            wget \
+            zlib1g-dev \
+    ; \
+# Install BamID
+    wget ${BAMID_URL}/${GIT_HASH}.zip; \
+    unzip ${GIT_HASH}.zip; \
+    \
+    cd VerifyBamID-${GIT_HASH}; \
+    mkdir build;  \
+    cd build; \
+    CC=$(which gcc) CXX=$(which g++) cmake ..; \
+    \
+    cmake; \
+    make; \
+    make test; \
+    \
+    cd ../../; \
+    mv VerifyBamID-${GIT_HASH}/bin/VerifyBamID .; \
+    rm -rf ${GIT_HASH}.zip VerifyBamID-${GIT_HASH} \
+    ; \
+# Install tini
+    wget https://github.com/krallin/tini/releases/download/$TINI_VERSION/tini -O /sbin/tini; \
+    chmod +x /sbin/tini \
+    ; \
+# Clean up cached files
+    apt-get clean && rm -rf /var/lib/apt/lists/*
+
+# Set tini as default entrypoint
+ENTRYPOINT [ "/sbin/tini", "--" ]
+```
```

Comment:
Might be good to add a final section on what to do if someone runs into questions about our Docker images or contributing a Docker image (i.e. File an Issue)